haystack/components/agents/agent.py

```diff
@@ -322,6 +335,10 @@ async def run_async(
         if self.system_prompt is not None:
             messages = [ChatMessage.from_system(self.system_prompt)] + messages
 
+        streaming_callback = select_streaming_callback(
+            init_callback=self.streaming_callback, runtime_callback=streaming_callback, requires_async=False
```

Comment:
This is in `run_async` right? If so this should be set to True
```suggestion
            init_callback=self.streaming_callback, runtime_callback=streaming_callback, requires_async=True
```